Leo's new ranges

OlivierBBB opened the pull request.

OlivierBBB committed:
feat(replays): added Leo's new ranges

OlivierBBB committed:
feat(replay): many more ranges

OlivierBBB committed:
fix(DeploymentInfo): two fixes

- used == instead of .equals(...)
- TX_SKIP has a subtlety ^^

OlivierBBB committed:
fix(hub): refined condition for exitDeploymentFromDeploymentInfoPointOfView

OlivierBBB committed:
spotless

OlivierBBB committed:
fix(tests): removed the @Disabled's

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -711,9 +711,27 @@ public void traceContextExit(MessageFrame frame) {
     Preconditions.checkArgument(
-        deploymentStatusOfBytecodeAddress()
-            == (messageFrame().getType() == MessageFrame.Type.CONTRACT_CREATION));
+        this.currentFrame().byteCodeAddress().equals(this.bytecodeAddress()));
```
I was using `==` rather than `.equals`

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -711,9 +711,27 @@ public void traceContextExit(MessageFrame frame) {
+       */
+      Preconditions.checkArgument(!deploymentStatusOfBytecodeAddress());
+    }
+
```
The only subtle point

letypequividelespoubelles commented on ranges_file.txt:
```diff
@@ -0,0 +1,194 @@
+5389571-5389577
```
Do we need to keep it ?

And for all replay json, if they all crashed the same error, maybe keep only one?

OlivierBBB replied:
We don't need this file, I will delete it.

OlivierBBB replied:
I will delete it later, it's something I discussed with @gusiri (the ability to provide custom `ranges.csv` files)

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -711,9 +711,27 @@ public void traceContextExit(MessageFrame frame) {
+    if (this.state.processingPhase != TX_SKIP) {
+      Preconditions.checkArgument(
+          deploymentStatusOfBytecodeAddress()
+              == (messageFrame().getType() == MessageFrame.Type.CONTRACT_CREATION),
```
.equals ?

OlivierBBB replied:
It's an `enum` comparison, no `.equals(...)` required.

OlivierBBB committed:
ras

OlivierBBB committed:
ras

OlivierBBB committed:
fix(call): we were passing the wrong caller !

The CallSection constructor happens in the current execution context and
it's the "accountAddress" that matters in terms of taking account
snapshots etc ...

OlivierBBB committed:
fix(call): better: trust Besu

OlivierBBB committed:
fix(call): extracted value uniformly

OlivierBBB committed:
ras

OlivierBBB committed:
ras

OlivierBBB committed:
fix(this): getting rid of this

OlivierBBB committed:
ras

OlivierBBB committed:
fix(create): moved "handleCreate" from tracePostExecution to the CreateSection

the stack only gets updated with the createe's address after the
deployment (and only if the deployment was successful) so that
tracePostExecution was necessarily wrong

also some renaming:
- newDeploymentAt                ==>  newDeploymentWithExecutionAt
- newDeploymentWithoutExecution  ==>  newDeploymentWitoutExecutionAt

OlivierBBB committed:
fix(selfdestruct): simplified the "fresh new dep#" part of SD

OlivierBBB committed:
fix: too many to list + a lot of cosmetic modifications (static imports ...)

OlivierBBB committed:
Merge branch 'arith-dev' into 1046-logs-for-the-first-range-5118361-5118389-exitdeploymentfromdeploymentinfopointofview

OlivierBBB committed:
fix: plenty of stuff + ras

# Conflicts:
#	arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/fragment/imc/mmu/opcode/ReturnFromDeployment.java
#	arithmetization/src/test/java/net/consensys/linea/zktracer/ReplayTests.java

OlivierBBB committed:
ras

OlivierBBB committed:
ras

OlivierBBB committed:
fix: fixed double deployment (didn't I fix this a while ago ?)

OlivierBBB committed:
fix(CREATE): fixed checkArgument for exitDeploymentFromDeploymentInfoPov

... there was an IRL Failure Condition F case for CREATE2 in this range
at absolute transaction number 178. This made a checkArgument fail. The
check is fixed (with an unsustainable hack that will have to be fixed.)

OlivierBBB committed:
ras: tests pass here

OlivierBBB committed:
ras: last remaing tests, if god wills it.

OlivierBBB committed:
spotless

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -301,12 +302,16 @@ private List<Module> precompileLimitModules() {
   private final List<Module> refTableModules;
 
+  /** fix this mess */
+  public boolean failureCondition_ArnoldPalmerAlert_ArnoldPalmerAlert_WhoWantsSomeArniePalmies =
```
WTF ?

OlivierBBB replied:
What's wrong? I blame Copilot 😎

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -620,29 +619,31 @@ public void traceContextEnter(MessageFrame frame) {
-          frame.getContractAddress(),
-          frame.getRecipientAddress(), // TODO: this is likely false
-          new Bytecode(frame.getCode().getBytes()),
+      // TODO: the
```
?

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -620,29 +619,31 @@ public void traceContextEnter(MessageFrame frame) {
-          this.transients.conflation().deploymentInfo().deploymentNumber(codeAddress),
-          codeDeploymentNumber,
-          isDeployment);
-      this.currentFrame().initializeFrame(frame); // TODO should be done in enter
```
// TODO should be done in enter

because we do it both if depth == 0 and if depth > 0

OlivierBBB replied:
I tried doing it for once for both but it breaks even our tests. I think it's related to entering the modules.

OlivierBBB replied:
I tried doing it for once for both but it breaks even our tests. I think it's related to entering the modules.@

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -685,35 +686,81 @@ public void traceContextExit(MessageFrame frame) {
+      } else {
+        checkArgument(!deploymentStatusOfBytecodeAddress());
+      }
+      // checkArgument(
```
comment to delete

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -806,25 +849,25 @@ public void tracePostExecution(MessageFrame frame, Operation.OperationResult ope
   }
 
+  public int callDataContextNumber(final boolean shouldCopyTxCallData) {
+    return shouldCopyTxCallData ? this.stamp() : 0;
```
is this boolean still needed ? I recall we finally create a callFrame for the tx callData, even if it's empty, n the case someone does a CALLDATACOPY, we need to have in the callFrame list (I fixed this in a past replayTest imho ...)

OlivierBBB replied:
We will not be copying zeros from an empty calldata array. It's done without reference to any context

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/romlex/RomLex.java:
```diff
@@ -120,15 +116,18 @@ public Bytes getCodeByMetadata(final ContractMetadata metadata) {
     return getChunkByMetadata(metadata).map(RomChunk::byteCode).orElse(Bytes.EMPTY);
   }
 
+  // TODO: it would maybe make more sense to only implement traceContextEnter
```
hum ... not sure. Because when we enter a context, ROMLEX is triggered by the call / create section. Here ROMLEX is triggered by the Besu hook, so it's quite different.

letypequividelespoubelles commented on arithmetization/src/test/java/net/consensys/linea/zktracer/ReplayTests.java:
```diff
@@ -199,84 +203,346 @@ void replayMultipleReplayFiles() {
+    replay("8718090.json.gz", false);
   }
 
+  // @Disabled
```
why all teg @Disabled are commented ? Better delete them no ?

OlivierBBB replied:
It's just easier to uncomment them again when they are already in the file. Running the full tests (with the replays) take 2.5 hours.

letypequividelespoubelles commented on arithmetization/src/test/java/net/consensys/linea/zktracer/ReplayTests.java:
```diff
@@ -199,84 +203,346 @@ void replayMultipleReplayFiles() {
+    replay("41640-41649.json.gz", false);
+  }
+
+  // Leo's range split up 5104800-5104883
```
could be nice to have a descriptive name of the test to know what was the issue, and let us have the possibility to run unit test to reduce the nb of replay tests. CI will be too long if not ...

OlivierBBB replied:
Let me see what I can do.

letypequividelespoubelles commented on testing/src/main/java/net/consensys/linea/testing/ToyExecutionEnvironmentV2.java:
```diff
@@ -147,7 +147,7 @@ private MainnetTransactionProcessor getMainnetTransactionProcessor(EVM evm) {
 
     final ContractCreationProcessor contractCreationProcessor =
-        new ContractCreationProcessor(evm, false, List.of(), 0);
+        new ContractCreationProcessor(evm, false, List.of(), 1);
```
What's the impact ?

OlivierBBB replied:
It was first noticed by @davepearce. It means that accounts created start out with a nonce of 1 (as in the EVM since some old hard fork) as opposed to an initial value of 0 (as was the case prior to said hardfork.)

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/romlex/ContractMetadata.java:
```diff
@@ -25,15 +26,16 @@
+    return new ContractMetadata(
+        address, hub.deploymentNumberOf(address), hub.deploymentStatusOf(address));
+  }
+
```
`canonical` should become the preferred method for creating `ContractMetadata`

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/romlex/ContractMetadata.java:
```diff
@@ -25,15 +26,16 @@
-    return new ContractMetadata(address, deploymentNumber, false);
-  }
-
   public static ContractMetadata underDeployment(final Address address, int deploymentNumber) {
```
Can we get rid of `underDeployment` by deferring it ?

OlivierBBB committed:
fix: removed some of this

OlivierBBB committed:
fix(hub): variable name + flattened exitDeploymentFromDeploymentInfoPov

OlivierBBB committed:
merge: arith-dev into working branch

OlivierBBB committed:
merge: arith-dev into working branch II

OlivierBBB committed:
fix(replays): Enabled the one disabled test

letypequividelespoubelles committed:
Merge branch 'arith-dev' into 1046-logs-for-the-first-range-5118361-5118389-exitdeploymentfromdeploymentinfopointofview

letypequividelespoubelles committed:
spotless after rebase

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
re fix rebase

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>
